numpy/core/src/npysort/quicksort.c.src

```diff
@@ -107,6 +111,7 @@ quicksort_@suff@(void *start, npy_intp num, void *NOT_USED)
                 *sptr++ = pi - 1;
                 pl = pi + 1;
             }
+            *psdepth++ = cdepth + 1;
```

Comment:
the confusion is probably in what each variable is.
the depth stack pointer is `psdepth` pointing to the top of the depth stack (name `depth` above)
`cdepth` is the value _on_ top the stack

     *psdepth++ = cdepth + 1

so here we increase the current depth by one, increase the stack pointer and put the new increased depth onto it.
may be rename the stack depth variable to `pdepth_stack` and the current value to `current_depth`?